mongoquery/__init__.py

```diff
@@ -54,31 +54,29 @@ def _extract(self, entry, path):
         else:
             return _Undefined()
 
-    def _get_value_at_path(self, obj, path):
-        is_dotted = True if path.find(".") != -1 else False
-        is_indexed = True if path.find("[") != -1 else False
-        if is_indexed:
-            path = path.replace("[", ".")
-            is_dotted = True
-        keys = " ".join(path.split(".")).split() if is_dotted else [ path ]
-        for k in keys:
-            if k.endswith(']'):
-                k = int(k[:-1])
-            if type(obj) is str:
-                raise KeyError("Invalid key '{}' in JSON path".format(k))
-            obj = obj[k]
-        return obj
-
     def _path_exists(self, operator, condition, entry):
-        try:
-            self._get_value_at_path(entry, operator)
-        except:
-            return not condition
+        if type(operator) is str:
```

Comment:
Probably better as something like isinstance(operator, BaseString) as this would trip over on python2 with unicode.